# no-raw-dom-access

âŒ Disallow raw DOM access (document.querySelector, getElementById, etc.) inside Alpine.js directives like @click, x-init, etc.

âœ… Recommend using Alpine's built-in x-ref and $refs instead.

ğŸ”§ Rule Details

This rule discourages raw DOM access inside Alpine.js directives and encourages using x-ref for cleaner, declarative code.

âŒ Incorrect

```html
<button @click="document.querySelector('#myInput').focus()">Click</button>
```

âœ… Correct

```html
<input x-ref="myInput">
<button @click="$refs.myInput.focus()">Click</button>
```